analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -125,9 +127,12 @@ private static void CheckForIsAssignableFrom(SonarSyntaxNodeReportingContext con
         {
             if (methodSymbol.Name == nameof(Type.IsAssignableFrom) && (argument as InvocationExpressionSyntax).IsGetTypeCall(context.SemanticModel))
             {
-                if (memberAccess.Expression is TypeOfExpressionSyntax)
+                if (memberAccess.Expression is TypeOfExpressionSyntax typeOf)
                 {
-                    ReportDiagnostic(context, MessageIsOperator, useIsOperator: true, shouldRemoveGetType: true);
+                    if (!(context.SemanticModel.GetTypeInfo(typeOf.Type).Type is INamedTypeSymbol { IsUnboundGenericType: true }))
```

Comment:
The result of previous comment will probably be a local static method that should get reused here too